File: hw4/Facts.java
// importing Scanner class from the Java library to be able to get user inputs
import java.util.Scanner;

// Very peculiar that even the main function has to go in a class
public class Facts {
  // defining the factorial function before the start of the main function
  // It is a recursive function that multiplies the original argument times argument-1 * argument-1-1 ... until the it passes argument 1 to the factorial function
  static double factorial(int arg) {
      if (arg > 1) {
        return arg*factorial(arg-1);
      } else {
        // once the argument hits 1, return a hardcoded int one since no need for further recursion
        return 1;
      }
    }  
  public static void main(String[] args) {
    // have to creat a new objected of scanner since Java doesn't support functions that are outside class definitions
    Scanner reader = new Scanner(System.in);
    // instantiate int n to any number but 0 since my loop logic will run infinitely until the user enters a 0
    int n = 1;
    while(true) {
      if(n != 0) {
        System.out.println("Enter a number you want the factorial of: (0 to exit)");
        n = reader.nextInt();
        Facts testobject = new Facts();
        System.out.println("n --------------- factorial(n)");
        System.out.println(n + "\t\t   " + testobject.factorial(n));
      } else {
        // If user enters "0", the program will exit
        break;
      }
    }
  }
  
}

File: hw4/facts.py
def factorial(arg):
  if arg > 0:
    return arg * factorial(arg-1)
  else:
    return 1



userinput = 1
while True:
  if userinput != 0:
    userinput = input("Enter a number you want the factorial of (0 to exit): \n\n\n")
    print("n --------------- factorial(n)\n%s                    %s") %(userinput, factorial(userinput))

    # print("n --------------- factorial(n)\n")
    # print("%s") %(userinput)
    # print("                    ")
    # print(factorial(userinput))
  else:
    break


# Once the user the factorial loop, it will display the square roots from 0 to 6
print("You exited the factorial loop, but here is the original FOR LOOP, which will run all the factorials from 0 to your number\n")
userinput = input("Enter a number you want the factorial of (will exit after displaying the factorials)\n")
print("n --------------- factorial(n)")
for n in range(1, userinput+1):
    print("%s \t\t    %s") %(n, factorial(n))